.github/workflows/fork-connector-tests.yml

```diff
@@ -0,0 +1,138 @@
+          curl -X POST \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github+json" \
+            "${{ github.api_url }}/repos/${{ github.repository }}/statuses/${{ github.event.pull_request.head.sha }}" \
+            -d '{
+              "state": "pending",
+              "context": "Fork Connector Tests",
+              "description": "Running connector tests in fork with BYO secrets...",
+              "target_url": "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+            }'
+
+      - name: Trigger connector tests in fork
+        id: trigger-tests
+        if: steps.get-app-token.outcome == 'success'
+        uses: the-actions-org/workflow-dispatch@v4
+        with:
+          token: ${{ steps.get-app-token.outputs.token }}
+          repo: ${{ steps.check-fork.outputs.fork-repo }}
+          workflow: connector-ci-checks.yml
+          ref: ${{ github.event.pull_request.head.ref }}
+          wait-for-completion: true
+          wait-for-completion-timeout: 2h
+          wait-for-completion-interval: 30s
+          inputs: |
+            {
+              "repo": "${{ steps.check-fork.outputs.fork-repo }}",
+              "gitref": "${{ github.event.pull_request.head.ref }}",
+              "pr": "${{ github.event.pull_request.number }}"
+            }
+
+      - name: Post success commit status
+        if: steps.trigger-tests.outcome == 'success' && steps.trigger-tests.outputs.workflow-conclusion == 'success'
+        run: |
+          curl -X POST \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github+json" \
+            "${{ github.api_url }}/repos/${{ github.repository }}/statuses/${{ github.event.pull_request.head.sha }}" \
+            -d '{
+              "state": "success",
+              "context": "Fork Connector Tests",
+              "description": "Connector tests passed in fork with BYO secrets",
+              "target_url": "${{ steps.trigger-tests.outputs.workflow-url }}"
+            }'
+
+      - name: Post failure commit status
+        if: steps.get-app-token.outcome == 'success' && (steps.trigger-tests.outcome == 'failure' || steps.trigger-tests.outputs.workflow-conclusion != 'success')
+        run: |
+          curl -X POST \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github+json" \
+            "${{ github.api_url }}/repos/${{ github.repository }}/statuses/${{ github.event.pull_request.head.sha }}" \
+            -d '{
+              "state": "failure",
+              "context": "Fork Connector Tests",
+              "description": "Connector tests failed in fork with BYO secrets",
+              "target_url": "${{ steps.trigger-tests.outputs.workflow-url || format(''{0}/{1}/actions/runs/{2}'', github.server_url, github.repository, github.run_id) }}"
+            }'
```

Comment:
Post success and post failure should be the same job with dynamic expressions used as needed for different inputs.